src/main/java/parser/TaskParser.java

```diff
@@ -0,0 +1,96 @@
+package parser;
+
+import exceptions.MorganException;
+import tasks.DeadlineTask;
+import tasks.EventTask;
+import tasks.Task;
+import tasks.ToDoTask;
+
+/**
+ * This is a parser.TaskParser class, which translates a task string
+ * into a Task object and vise versa.
+ */
+public class TaskParser {
+    public static final String DELIMITER = "¬";
+    private static final int TASK_INDEX = 0; // Task type
+    private static final int STATUS_INDEX = 1; // Task status
+    private static final int DESCRIPTION_INDEX = 2; // Task description
+    private static final int DATETIME_INDEX = 3; // Task date/time (if applicable)
+    private static final String TAMPERED_ERROR = "Storage data has been tampered.";
+
+    /**
+     * Translates a task into its storage string representation.
+     * @param task The task to be parsed.
+     * @return The storage string representation of the task.
+     */
+    public String encode(Task task) {
```

Comment:
I like your usage of your conditional statements here - very clean and understandable.